pkg/capacity/capacity.go

```diff
@@ -551,13 +561,17 @@ func (c *Controller) syncCapacity(ctx context.Context, item workItem) error {
 		// would race with receiving that object through the event handler. In the unlikely
 		// scenario that we end up creating two objects for the same work item, the second
 		// one will be recognized as duplicate and get deleted again once we receive it.
-	} else if capacity.Capacity.Value() == quantity.Value() {
+	} else if capacity.Capacity.Value() == quantity.Value() &&
+		(c.owner == nil || c.isOwnedByUs(capacity)) {
```

Comment:
Should this be `isManaged()`? If a driver running in binary mode manages an object which is unchanged, it should still skip right?

Reply:
This really is an ownerref check. It covers the scenario where a driver was deployed without ownerref and then gets redeployed with ownerref. The existing objects then need to be updated even when the capacity is the same, which is achieved here by falling through to the final else clause.

I added "same capacity %v *and correct owner*" to the log message to make this more clear.